src/server/acl/acl_family_test.cc

```diff
@@ -275,4 +275,52 @@ TEST_F(AclFamilyTest, TestDryRun) {
   EXPECT_THAT(resp, ErrArg("ERR User: kostas is not allowed to execute command: SET"));
 }
 
+TEST_F(AclFamilyTest, AclGenPassTooManyArguments) {
+  TestInitAclFam();
+
+  auto resp = Run({"ACL", "GENPASS", "1", "2"});
+  EXPECT_THAT(resp.GetString(),
+              "ERR Unknown subcommand or wrong number of arguments for 'GENPASS'. Try ACL HELP.");
+}
+
+TEST_F(AclFamilyTest, AclGenPassOutOfRange) {
+  TestInitAclFam();
+
+  std::string expectedError =
+      "ERR ACL GENPASS argument must be the number of bits for the output password, a positive "
+      "number up to 4096";
+
+  auto resp = Run({"ACL", "GENPASS", "-1"});
+  EXPECT_THAT(resp.GetString(), expectedError);
+
+  resp = Run({"ACL", "GENPASS", "0"});
+  EXPECT_THAT(resp.GetString(), expectedError);
+
+  resp = Run({"ACL", "GENPASS", "4097"});
+  EXPECT_THAT(resp.GetString(), expectedError);
+}
+
+TEST_F(AclFamilyTest, AclGenPass) {
+  auto resp = Run({"ACL", "GENPASS"});
+  auto actualPassword = resp.GetString();
+
+  // should be 256 bits or 64 bytes in hex
+  //  EXPECT_THAT(actualPassword.length(), 64);
+
+  // 1 bits - 4 bits should all produce a single hex character
+  for (int i = 1; i <= 4; i++) {
+    resp = Run({"ACL", "GENPASS", std::to_string(i)});
+    EXPECT_THAT(resp.GetString().length(), 1);
+  }
+  // 5 bits - 8 bits should all produce two hex characters
+  for (int i = 5; i <= 8; i++) {
+    resp = Run({"ACL", "GENPASS", std::to_string(i)});
+    EXPECT_THAT(resp.GetString().length(), 2);
+  }
+
+  // and the pattern continues
+  resp = Run({"ACL", "GENPASS", "9"});
+  EXPECT_THAT(resp.GetString().length(), 3);
```

Comment:
9 bits and the next multiple of 4 for 9 is 12. (therefore the size of the string should be 4)

Reply:
I believe you are mistaken here, @kostasrim. 9+3 /4 = 3

Also, output from Redis:
```
127.0.0.1:6379> acl genpass 9
"327"
```

Reply:
Yes you are right!

Reply:
and also this is doing the same thing `(random_bits + 3) / 4`, nice!